src/components/Books.js

```diff
@@ -1,33 +1,36 @@
-import { useSelector } from 'react-redux';
+import { useSelector, useDispatch } from 'react-redux';
+import { useEffect } from 'react';
+import { fetchBooks } from '../redux/books/booksSlice';
 import Book from './Book';
 import AddNewBook from './AddNewBook';
 
-const Books = () => {
-  const { books } = useSelector((state) => state.books);
+const BookList = () => {
+  const dispatch = useDispatch();
+  const { booksCollection, status } = useSelector((store) => store.books);
+
+  useEffect(() => {
+    if (status !== 'idle') return;
+    dispatch(fetchBooks());
+  }, [booksCollection, status, dispatch]);
 
```

Comment:
A Good Way to stop infinite rendering (loop): @astroboyReloaded 
The dependencies array in the useEffect hook determines when the effect should be re-run. If you include an async function in the dependencies array, it will be considered a new reference on each render. Since functions are reference types in JavaScript, this will cause the effect to be re-executed on every render, leading to an infinite loop. so please remove [booksCollection, status, dispatch]. You have a better way to  use  [dispatch]. 

- I can give you a suggestion, I see you wanted to download all data from API , then render the data on UI.
I found it easy, if you add this useEffect with empty array (`[ ]`). But the location can be inside `App.js`;
-  This means , whenever user starts using app / refresh the page, data will be downloaded